nemo/collections/cv/modules/trainables/generic_image_encoder.py

```diff
@@ -0,0 +1,202 @@
+                self._feature_map_height = 7
+                self._feature_map_width = 7
+                self._feature_map_depth = 512
+
+            else:
+                # Use the whole model, but "reshape"/reinstantiate the last layer ("FC6").
+                self._model.classifier._modules['6'] = torch.nn.Linear(4096, self._output_size)
+
+        elif self._model_type == 'densenet121':
+            # Get densenet121
+            self._model = models.densenet121(pretrained=pretrained)
+
+            if self._return_feature_maps:
+                raise ConfigurationError("'densenet121' doesn't support 'return_feature_maps' mode (yet)")
+
+            # Use the whole model, but "reshape"/reinstantiate the last layer ("FC6").
+            self._model.classifier = torch.nn.Linear(1024, self._output_size)
+
+        elif self._model_type == 'resnet152':
+            # Get resnet152
+            self._model = models.resnet152(pretrained=pretrained)
+
+            if self._return_feature_maps:
+                # Get all modules exluding last (avgpool) and (fc)
+                modules = list(self._model.children())[:-2]
+                self._model = torch.nn.Sequential(*modules)
+
+                # Remember the output feature map dims.
+                self._feature_map_height = 7
+                self._feature_map_width = 7
+                self._feature_map_depth = 2048
+
+            else:
+                # Use the whole model, but "reshape"/reinstantiate the last layer ("FC6").
+                self._model.fc = torch.nn.Linear(2048, self._output_size)
+
+        elif self._model_type == 'resnet50':
+            # Get resnet50
+            self._model = models.resnet50(pretrained=pretrained)
+
+            if self._return_feature_maps:
+                # Get all modules exluding last (avgpool) and (fc)
+                modules = list(self._model.children())[:-2]
+                self._model = torch.nn.Sequential(*modules)
+
+                # Remember the output feature map dims.
+                self._feature_map_height = 7
+                self._feature_map_width = 7
+                self._feature_map_depth = 2048
+
+            else:
+                # Use the whole model, but "reshape"/reinstantiate the last layer ("FC6").
+                self._model.fc = torch.nn.Linear(2048, self._output_size)
+
+    @property
+    @add_port_docs()
+    def input_ports(self):
+        """
+        Returns definitions of module input ports.
+        """
+        return {
+            "inputs": NeuralType(
+                axes=(
+                    AxisType(kind=AxisKind.Batch),
+                    AxisType(kind=AxisKind.Channel, size=3),
+                    AxisType(kind=AxisKind.Height, size=224),
+                    AxisType(kind=AxisKind.Width, size=224),
+                ),
+                elements_type=ImageType(),
+                # TODO: actually encoders pretrained on ImageNet require special image normalization.
```

Comment:
I think this is fine. Special normalized image types could inherit from ImageType